### PR TITLE
Flush methods on container close

### PR DESCRIPTION
**Pull request by mihran113**
Added `flush` and `flush_wal` on container close and changed db write options based on previous discussions

### CODE REVIEW

File: aim/storage/rockscontainer.py [hunk 1]
```diff
@@ -106,12 +107,18 @@ def finalize(self, *, index: Container):
...
+        self._db.flush_wal()
+
     def close(self):
         """Close all the resources."""
         if self._lock is not None:
             self._lock.release()
             self._lock = None
         if self._db is not None:
+            if not self.read_only:
+                self._db.flush()
```
Comment (alberttorosyan) [inline aim/storage/rockscontainer.py:120]: I guess we have to keep db locked while flushes are running.

File: aim/storage/rockscontainer.py [hunk 2]
```diff
@@ -106,12 +107,18 @@ def finalize(self, *, index: Container):
         self._progress_path.unlink()
         self._progress_path = None
 
+        self._db.flush()
```
Comment (alberttorosyan) [inline aim/storage/rockscontainer.py:110]: as with the locks, db is considered as 'in progress' unless flushed

**Review by alberttorosyan — CHANGES_REQUESTED** [review]
@mihran113 jan, left comments below

File: aim/storage/rockscontainer.py (hunk 2 above)
Comment (mihran113) [reply aim/storage/rockscontainer.py:110]: yup, legit notices, will change ✅

**Review by alberttorosyan — APPROVED** [review]
lgtm

**Comment by mihran113** [comment]
resolves #963

**Pull request merged**